create-plugin-update/pm.sh

```diff
@@ -15,36 +15,31 @@ install_pnpm_if_not_present() {
 }
 
 # Detect the package manager
+# and the args to run installation
+# and the exec command to run create-plugin
 if [ -f yarn.lock ]; then
 	pm="yarn"
+	pmi=()
+	pmx=("yarn" "create" "@grafana/plugin")
 elif [ -f pnpm-lock.yaml ]; then
 	install_pnpm_if_not_present
 	pm="pnpm"
+	pmi=("--no-frozen-lockfile")
+	pmx=("pnpm" "dlx" "@grafana/create-plugin@latest")
 elif [ -f package-lock.json ]; then
 	pm="npm"
+	pmi=()
+	pmx=("npx" "-y" "@grafana/create-plugin@latest")
 else
 	echo "No recognized package manager found in this project."
 	exit 1
 fi
 
-# Detect the exec command to run create-plugin update
-if [ -f yarn.lock ]; then
-  pmx=("yarn" "create" "@grafana/plugin")
-elif [ -f pnpm-lock.yaml ]; then
-  install_pnpm_if_not_present
-  pmx=("pnpm" "dlx" "@grafana/create-plugin@latest")
-elif [ -f package-lock.json ]; then
-  pmx=("npx" "-y" "@grafana/create-plugin@latest")
-else
-  echo "No recognized package manager found in this project."
-  exit 1
-fi
-
 # Run the provided command with the detected package manager
 if [ "$1" = "install" ]; then
-  echo "Running '$1' with $pm..."
-	"$pm" install
+	echo "Running '$1' with $pm..."
+	"$pm" install "${pmi[@]}"
 elif [ "$1" = "update" ]; then
-  echo "Running '$1' with ${pmx[0]}..."
+	echo "Running '$1' with ${pmx[0]}..."
```

Comment:
As there are only two commands that can be run (install and update) it might be easier to reason about this code if we only have `pmi` and `pmx` and update the logic that sets the command above to run to be something like:

```
pmi=("yarn" "install" "other" "args")
pmx=("yarn" "create" "@grafana/plugin" "update" "other" "args")
```

Then here we'd call the command rather than the interpolation we've currently got going on.